engine/namegen.ml

```diff
@@ -308,6 +308,13 @@ let visible_ids sigma (nenv, c) =
       | _ -> ids
       in
       accu := (gseen, vseen, ids)
+  | exception Not_found when !Flags.in_debugger ->
```

Comment:
This would be an exception raised by EConstr.kind, but AFAICT that doesn't raise Not_found. Do you have an example where this catch is useful?

Reply:
I thought `whd_evar` could raise `Not_found` on a bad `sigma`, but you're indeed right, it just behaves in this case as if an undefined evar (`existential_opt_value`).